tests/integration/components/utils/templated-input-group-test.ts

```diff
@@ -0,0 +1,73 @@
+import { hbs } from 'ember-cli-htmlbars';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupIntl } from 'ember-intl/test-support';
+import { click, render, typeIn } from '@ember/test-helpers';
+import sinon from 'sinon';
+
+module('Integration | Component | utils/templated-input-group', function (hooks) {
+  setupRenderingTest(hooks);
+  setupIntl(hooks);
+
+  hooks.beforeEach(function () {
+    this.title = 'Input Title';
+    this.value = '';
+    this.id = 'input-id';
+    this.variables = ['InstagramUsername', 'TiktokUsername'];
+    this.onChange = sinon.spy();
+  });
+
+  test('It renders', async function (assert) {
+    await render(
+      hbs`<Utils::TemplatedInputGroup @title={{this.title}} @value={{this.value}} @variables={{this.variables}} @onChange={{this.onChange}} @id={{this.id}} />`
+    );
+    assert.dom('[data-control-name="templated-input-group-insert-variable-link"]').exists();
+    assert.dom('[data-control-name="templated-input-group-input-container"]').exists();
+    assert.dom('#input-id').exists();
+  });
+
+  module('Dropdown variables list', () => {
+    test('Dropdown variables list is opened on click on link', async function (assert) {
+      await render(
+        hbs`<Utils::TemplatedInputGroup @title={{this.title}} @value={{this.value}} @variables={{this.variables}} @onChange={{this.onChange}} @id={{this.id}} />`
+      );
+      assert.dom('.upf-floating-menu--hidden').exists();
+      assert.dom('.upf-floating-menu--visible').doesNotExist();
+
+      await click('[data-control-name="templated-input-group-insert-variable-link"]');
+      assert.dom('.upf-floating-menu--hidden').doesNotExist();
+      assert.dom('.upf-floating-menu--visible').exists();
+      assert.dom('.upf-floating-menu--visible').hasText('InstagramUsername TiktokUsername');
+    });
+
+    test('Dropdown variables list is opened when {{ is typed in input', async function (assert) {
+      await render(
+        hbs`<Utils::TemplatedInputGroup @title={{this.title}} @value={{this.value}} @variables={{this.variables}} @onChange={{this.onChange}} @id={{this.id}} />`
+      );
+
+      await typeIn('.upf-input', '{{', { delay: 0 });
+      assert.dom('.upf-floating-menu--visible').exists();
+    });
+
+    test('Clicking on an item in dropdown closes it and updates input value', async function (assert) {
+      await render(
+        hbs`<Utils::TemplatedInputGroup @title={{this.title}} @value={{this.value}} @variables={{this.variables}} @onChange={{this.onChange}} @id={{this.id}} />`
+      );
+
+      await click('[data-control-name="templated-input-group-insert-variable-link"]');
+      await click('.upf-floating-menu__item:nth-child(1)');
+      assert.dom('.upf-floating-menu--hidden').exists('Item selected, dropdown closed');
+      assert.ok(this.onChange.calledWith('{{InstagramUsername}}', true));
```

Comment:
Use more precise check `calledWith` -> `calledOnceWithExactly`
Also, add a case in your tests when the `onChange` returns a no valid value

```ts
assert.ok(this.calledOnceWithExactly.calledWith('{{Ramdom}}', false));
```